packages/db-model/src/model.ts

```diff
@@ -72,253 +70,217 @@ export function createBaseModel(resolver?: ConnectionResolver) {
     }
 
     /**
-     * Generic query builder.
+     * Finds a single record based on the params.
      *
-     * @param {(qb: Knex | Transaction) => QueryBuilder} callback
-     * @param {Transaction} [trx]
-     * @returns {Promise<T[]>}
+     * @param {object} [params={}]
+     * @param {Knex.Transaction} trx
+     * @param {boolean} shouldFailIfNotFound
+     * @param {Function} callback
+     * @returns {Promise<any>}
      */
-    public static async buildQuery<T>(
-      callback: (qb: Knex | Transaction) => QueryBuilder,
-      trx?: Transaction
-    ): Promise<T[]> {
-      const qb = db.queryBuilder(this.getConnection(), trx);
-      const result = await callback(qb);
-
-      return object.toCamelCase<T[]>(result);
+    public static findFirst(
+      params: object = {},
+      shouldFailIfNotFound?: boolean,
+      callback?: any,
+      trx?: Knex.Transaction
+    ): Promise<any> {
+      return db
+        .findFirst(this.getConnection(), this.table, params, this.defaultOrderBy, callback, trx)
+        .then(([result]) => {
+          if (!result && shouldFailIfNotFound) throw new ModelNotFoundError('No row found for the given identifier.');
```

Comment:
Why should this method throw an error if the record is not found?This wrapper should have nothing to do with throwing an exception or not when the record exist or not.  This is something logical not generic.
Therefore findFirst should return either result or null, the client code should decide whether to throw an exception or not